pypeit/core/flexure.py

```diff
@@ -72,29 +72,32 @@ def spat_flexure_shift(sciimg, slits, debug=False, maxlag=20):
     mean_sci, med_sci, stddev_sci = stats.sigma_clipped_stats(_sciimg[onslits])
     thresh =  med_sci + 5.0*stddev_sci
     corr_sci = np.fmin(_sciimg.flatten(), thresh)
-
     lags, xcorr = utils.cross_correlate(corr_sci, corr_slits, maxlag)
     xcorr_denom = np.sqrt(np.sum(corr_sci*corr_sci)*np.sum(corr_slits*corr_slits))
     xcorr_norm = xcorr / xcorr_denom
     # TODO -- Generate a QA plot
+
     tampl_true, tampl, pix_max, twid, centerr, ww, arc_cont, nsig \
             = arc.detect_lines(xcorr_norm, sigdetect=3.0, fit_frac_fwhm=1.5, fwhm=5.0,
                                cont_frac_fwhm=1.0, cont_samp=30, nfind=1, debug=debug)
     # No peak? -- e.g. data fills the entire detector
     if len(tampl) == 0:
-        msgs.warn('No peak found in spatial flexure.  Assuming there is none..')
-#        if debug:
-#            embed(header='68 of flexure')
+        msgs.warn('No peak found in spatial flexure.  Assuming there is none...')
+        
         return 0.
-
+    
     # Find the peak
     xcorr_max = np.interp(pix_max, np.arange(lags.shape[0]), xcorr_norm)
     lag_max = np.interp(pix_max, np.arange(lags.shape[0]), lags)
+    #If xcorr is broad but has a defined peak, use that since detect_lines might fail
+    # NOTE: This is definitely a kludge, but may be necessary for NIRSPEC - ASC
+    if xcorr_max < np.max(xcorr_norm):
+        lag_max = [lags[lags > 0][np.argmax(xcorr_norm[lags > 0])]]
```

Comment:
This was a workaround but is no longer necessary due to the implementation of the manual correction. In retrospect it is possible this would chase spurious signals so I've removed it.